client/lib/track-form/index.jsx

```diff
@@ -0,0 +1,50 @@
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import { noop, union } from 'lodash';
+
+export const trackForm = WrappedComponent => {
+	class TrackFormComponent extends Component {
+		state = {
+			dirtyFields: [],
+			fields: {}
+		};
+
+		updateFields = ( fields, callback = noop ) => {
```

Comment:
I don't know that you really need a default value here since `setState` will simply ignore the undefined second argument.